pkg/grpc/server.go

```diff
@@ -147,6 +152,16 @@ func (a *apiImpl) Register(services ...APIService) {
 	a.apiServices = append(a.apiServices, services...)
 }
 
+func (a *apiImpl) Stop() *concurrency.Signal {
```

Comment:
I am not 100% sure on this one. What do you think about making this call blocking? 🤔 
It has the advantage that there is no signal handling, and in my assumption a stop of the grpc server mostly is an operation which is only done once when the process / connection terminates.

Reply:
I like the idea, let me give it a try and see how it looks.

The only reason I used this signature was to keep it symmetric with `Start()`, which also returns a signal. But you have a point, making this call blocking will simplify the concurrent code.